src/homework1.py

```diff
@@ -0,0 +1,131 @@
+import itertools
+import random
+import re
+
+# 1 line: Output
+
+print('Hello, world!')
+
+# 2 lines: Input, assignment
+
+name = input('What is your name?\n')
+print('Hi, %s.' % name)
+
+# 3 lines: For loop, built-in enumerate function, new style formatting
+
+friends = ['john', 'pat', 'gary', 'michael']
+for i, name in enumerate(friends):
+    print("iteration {iteration} is {name}".format(iteration=i, name=name))
+
+# 4 lines: Fibonacci, tuple assignment
+
+parents, babies = (1, 1)
+while babies < 100:
+    print('This generation has {0} babies'.format(babies))
+    parents, babies = (babies, parents + babies)
+
+# 5 lines: Functions
```

Comment:
Tasks should be separated with either one or two blank lines everywhere (it's not accurate to separate it with one line in one task and with two in another)